test/Serialization/access-level-import-dependencies.swift

```diff
@@ -95,4 +95,24 @@ import PrivateDep
 // RUN: %target-swift-frontend -typecheck %t/ClientOfPublic.swift -I %t \
 // RUN:   -Rmodule-loading 2>&1 | %FileCheck -check-prefix=VISIBLE-DEP %s
 // RUN: %target-swift-frontend -typecheck %t/ClientOfNonPublic.swift -I %t \
+// RUN:   -Rmodule-loading 2>&1 | %FileCheck -check-prefix=HIDDEN-DEP %s
+
+/// With testable imports, transitive dependencies are required.
+//--- TestableClientOfPublic.swift
+@testable import PublicDep
+
+//--- TestableClientOfNonPublic.swift
+@testable import PackageDep // expected-error {{missing required module 'HiddenDep'}}
+@testable import InternalDep // expected-error {{missing required module 'HiddenDep'}}
```

Comment:
Might be worth adding a test with `@_exported` import as well?

Reply:
I've added a test about it. From the commit message:

A scenario where module A `@testable` imports module B, which `@_exported` imports module C. In this case, non-public imports of B are loaded, but not those of C. This is a limitation of the current implementation and could be improved upon in the future.